rate.go

```diff
@@ -1,116 +1,237 @@
 package throttled
 
 import (
-	"math"
-	"net/http"
-	"strconv"
+	"fmt"
 	"time"
 )
 
-// Static check to ensure that rateLimiter implements Limiter.
-var _ Limiter = (*rateLimiter)(nil)
+const (
+	// Maximum number of times to retry SetIfNotExists/CompareAndSwap operations
+	// before returning an error.
+	maxCASAttempts = 10
```

Comment:
Thanks for documenting `maxCASAttempts` this time around :) I have a feeling that this will help somebody at some point.